pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -192,6 +204,28 @@ func isPodManagedContainer(cinfo *cadvisorapiv2.ContainerInfo) bool {
 	return managed
 }
 
+// getcadvisorPodInfoFromPodUID returns a pod cgroup information based on the podUID
+// In ContainerInfo data struct, the key for pod cgroup is normally one of the following format (<podUID> is the string of the pod UID)
+// "/kubepods/pod<podUID>"
+// "/kubepods/besteffort/pod<podUID>"
+// "/kubepods/burstable/pod<podUID>"
+// "/kubepods/guaranteed/pod<podUID>"
+func getcadvisorPodInfoFromPodUID(podUID types.UID, infos map[string]cadvisorapiv2.ContainerInfo) *cadvisorapiv2.ContainerInfo {
+	for key, info := range infos {
+		if strings.HasPrefix(key, "/") {
+			key = key[1:]
+		}
+		if strings.HasSuffix(key, "/") {
+			key = key[:len(key)-1]
+		}
+		keyparts := strings.Split(key, "/")
+		if len(keyparts) > 1 && strings.HasSuffix(keyparts[len(keyparts)-1], string(podUID)) {
```

Comment:
nit: use the `path` package. Useful methods might be `Clean` and `Split` or `Match`